assembly/token/token.contract.ts

```diff
@@ -99,22 +112,34 @@ export class TokenContract extends Contract {
      * @param memo - the memo string to accompany the transaction.
      */
     @action("transfer")
-    transfer(from: Name, to: Name, quantity: Asset, memo: string): void {
-        check(from != to, "cannot transfer to self");
+    transfer(
+        from: Name,
+        to: Name,
+        quantity: Asset,
+        memo: string
+    ): void {
+        // ? Why do we need memo here if it is not used anywhere in the code? 
+        check(memo.length <= 256, "memo has more than 256 bytes");
+
         requireAuth(from);
+
+        check(from != to, "cannot transfer to self");
```

Comment:
don't change the order of `check` statements in this file

Reply:
Why the order is important?